src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -9049,6 +9049,130 @@ void M(in string xIn)
+                //         Missing(F(x = null)); // 2
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(7, 23),
+                // (8,9): warning CS8602: Possible dereference of a null reference.
+                //         x.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(8, 9),
+                // (10,9): error CS0103: The name 'Missing' does not exist in the current context
+                //         Missing(F(x = this)); // 4
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "Missing").WithArguments("Missing").WithLocation(10, 9),
+                // (10,23): error CS0026: Keyword 'this' is not valid in a static property, static method, or static field initializer
+                //         Missing(F(x = this)); // 4
+                Diagnostic(ErrorCode.ERR_ThisInStaticMeth, "this").WithLocation(10, 23)
+                );
+        }
+
+        [Fact]
+        [WorkItem(29916, "https://github.com/dotnet/roslyn/issues/29916")]
+        public void PassingParameters_UnknownMethod_AffectingState()
+        {
+            CSharpCompilation c = CreateCompilation(new[] { @"
+class C
+{
+    object F(object x)
+    {
+        Missing(
+            F(x = null) /*warn*/,
+            x.ToString() /*warn*/,
+            F(x = this),
+            x.ToString());
+        return x;
+    }
+}
+" }, options: WithNonNullTypesTrue());
+
+            c.VerifyDiagnostics(
+                // (6,9): error CS0103: The name 'Missing' does not exist in the current context
+                //         Missing(
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "Missing").WithArguments("Missing").WithLocation(6, 9),
+                // (7,15): warning CS8604: Possible null reference argument for parameter 'x' in 'object C.F(object x)'.
+                //             F(x = null) /*warn*/,
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "x = null").WithArguments("x", "object C.F(object x)").WithLocation(7, 15),
+                // (7,19): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //             F(x = null) /*warn*/,
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(7, 19),
+                // (8,13): warning CS8602: Possible dereference of a null reference.
+                //             x.ToString() /*warn*/,
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(8, 13)
+                );
+        }
+
+        [Fact]
+        [WorkItem(29916, "https://github.com/dotnet/roslyn/issues/29916")]
+        public void PassingParameters_UnknownMethod_AffectingConditionalState()
+        {
+            CSharpCompilation c = CreateCompilation(new[] { @"
+class C
+{
+    int F(object x)
+    {
+        if (G(F(x = null)))
```

Comment:
>x = null [](start = 16, length = 8)

`x == null` in `WhenTrue` and `WhenFalse`. Consider testing where the nullability is different in the two states. Perhaps:
```c#
if (F(x) && F(x = null))
    x.ToString(); // ok
else
    x.ToString(); // warn
```